devdoc/docs/ContributeDocs.md

```diff
@@ -0,0 +1,394 @@
+# Contributing Documentation to the HPCC Systems Platform Project
+
+This document is intended for anyone that wants to contribute
+documentation to our project. The first audience is platform developers,
+so we can streamline the process of documenting new features. However,
+these guidelines apply to anyone who wants to contribute to any of our
+documentation (Language Reference, Programmer’s Guide, etc.).
+
+This set of guidelines should help you understand the information needed
+to create sufficient documentation for any new feature you add. The good
+news is that we are all here to help you and support your writing
+efforts. We will help by advising you along the way, and by reviewing
+and editing your submissions.
+
+## Documenting a New Software Feature--Required and Optional Components
+
+When you create a new feature or function, clear documentation is
+crucial for both internal teams and external users. You worked hard on
+the feature, so it deserves proper notice and usage.
+
+We are an open-source project, so contributions are always welcome,
+including documentation.
+
+You can contribute on many levels:
+
+1.  Developer Notes
+
+2.  End user “Readmes” in the form of MD files in the GitHub repository
+
+3.  Blogs
+
+4.  Formal documentation
+
+Regardless of the form you are planning to deliver, here are the
+required and optional components to include in a document.
+
+> **Tip**: VS Code is very good at editing MD files. There is a Preview
+> panel available to be able to see the rendered form.
```

Comment:
If there is something special that we need to do in VS Code to enable the MD editing, mention it.  If there isn't then  ignore this comment

Reply:
@stuartort Good question. MD support is built-in. There is an extension for better Mermaid diagram preview, but that is an advanced topic to consider later.